distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/cometd/CometDClient.java

```diff
@@ -177,8 +179,46 @@ public void subscribe(String channel) {
     }
 
     /**
-     * Gets the list of messages received on a given channel.
+     * Returns an action url from the messages of a channel if it contains the search criteria string.
      *
+     * @param searchCriterion a string that will be searched for in the messages.
+     * @param title           the title of the action we want to find.
+     * @return The desired action url if a match for the search criterion is found in the messages.
+     * Otherwise returns an empty string.
+     * @throws Exception
+     */
+    public String findActionUrlBySearchCriterion(String searchCriterion, String title)
+            throws Exception {
+
+        List<String> messages = getAllMessages();
+
+        Optional<String> foundString = messages.stream()
+                .filter(query -> query.contains(searchCriterion))
+                .findAny();
+
+        LOGGER.debug("Data found using search criterion: " + foundString.toString());
+
+        String desiredUrl = "";
+
+        if (foundString.isPresent()) {
+
+            JsonPath path = JsonPath.from(foundString.get());
+
+            //When the url of an action is needed, it can be found by searching for the action's title
+            desiredUrl = path.getString(
+                    "data.results[0].metacard.actions.find {it.title=='" + title + "'}.url");
```

Comment:
It would be more efficient to make this a static final parametrized search string:
```java
private static final String FIND_ACTION_URL_PATTERN = "data.results[0].metacard.actions.find {it.title=='%s'}.url";
...
desiredUrl = path.getString(String.format(FIND_ACTION_URL_PATTERN, title));
```